src/Compilers/VisualBasic/Portable/Compilation/VisualBasicCompilation.vb

```diff
@@ -708,6 +708,33 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 eventQueue:=eventQueue)
         End Function
 
+        Friend Overrides Sub SerializePdbEmbeddedCompilationOptions(builder As BlobBuilder)
+            WriteValue(builder, CompilationOptionNames.Checked, Options.CheckOverflow.ToString())
+
+            ' LanguageVersion should already be mapped to an effective version at this point
+            Debug.Assert(LanguageVersion.MapSpecifiedToEffectiveVersion() = LanguageVersion)
+            WriteValue(builder, CompilationOptionNames.LanguageVersion, LanguageVersion.ToDisplayString())
+
+            WriteValue(builder, CompilationOptionNames.Strict, Options.OptionStrict.ToString())
+
+            If Options.ParseOptions IsNot Nothing Then
+                Dim preprocessorStrings = Options.ParseOptions.PreprocessorSymbols.Select(Function(p)
+                                                                                              If (p.Value Is Nothing) Then
+                                                                                                  Return p.Key
+                                                                                              End If
+
+                                                                                              Return p.Key + "=" + p.Value.ToString()
+                                                                                          End Function)
+                WriteValue(builder, CompilationOptionNames.Define, String.Join(",", preprocessorStrings))
+            End If
+        End Sub
+
+        Private Sub WriteValue(builder As BlobBuilder, key As String, value As String)
```

Comment:
We have this logic duplicated in both of the Implementations. I know it's pretty simple, but has to the be the same to read it back. Would it make more sense to have the implementations return a (string, string) pair of key/value, and have the common compilation do the actual writing?

Reply:
From a signature point of view, the method signatures are more clear that they are writing to PDB blob. Maybe a common helper would be useful to write kvp?

Reply:
Eh, I think the logic is simple enough its fine to duplicate, just wondered if it would be clearer to keep it in a single place.

---
In reply to: [436125201](https://github.com/dotnet/roslyn/pull/44373#discussion_r436125201) [](ancestors = 436125201)